packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/utils/create_helpers.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Logger } from '@kbn/logging';
+import type {
+  IKibanaMigrator,
+  SavedObjectsSerializer,
+} from '@kbn/core-saved-objects-base-server-internal';
+import type {
+  ISavedObjectTypeRegistry,
+  SavedObjectsExtensions,
+} from '@kbn/core-saved-objects-server';
+import {
+  RepositoryHelpers,
+  CommonHelper,
+  EncryptionHelper,
+  ValidationHelper,
+  PreflightCheckHelper,
+  SerializerHelper,
+  MigrationHelper,
+} from '../apis/helpers';
+import type { RepositoryEsClient } from '../repository_es_client';
+import { CreatePointInTimeFinderFn } from '../point_in_time_finder';
+
+interface CreateRepositoryHelpersOptions {
+  index: string;
+  client: RepositoryEsClient;
+  typeRegistry: ISavedObjectTypeRegistry;
+  serializer: SavedObjectsSerializer;
+  migrator: IKibanaMigrator;
+  logger: Logger;
+  extensions?: SavedObjectsExtensions;
+  createPointInTimeFinder: CreatePointInTimeFinderFn;
+}
+
+export const createRepositoryHelpers = ({
+  logger,
+  extensions,
+  index,
+  client,
+  typeRegistry,
+  serializer,
+  migrator,
+  createPointInTimeFinder,
+}: CreateRepositoryHelpersOptions): RepositoryHelpers => {
```

Comment:
NIT: extracted from the repository

Reply:
At some point, these helpers are gonna get very nested. It's already the case with the `commonHelpers` that are now nested in here. I'm all for making life easier and reducing code but we need to be careful of taking it too far to onboard new folks.

Reply:
I totally agree. It's hard to get the cursor to the correct position here and not going too far... Especially given the codebase we're talking about